### ISSUE TITLE
Exhaust error is not handled 

### ISSUE THREAD

**Issue by MariamKhalatova** [Green-Software-Foundation/if#698]
## Description of the Error

On the branch `dump-logs-error`, exhaust error is not handled with current catch mechanism.
<img width="863" alt="Screenshot 2024-05-06 at 1 35 45 AM" src="https://github.com/Green-Software-Foundation/if/assets/109615222/b21931c2-5c6a-4284-8eed-f16e3ba94bc2">

## Expected Behaviour

Exhaust error to be handled by catch block. Logger should mark it as an error, and print it in red.

## Actual Behaviour

The process exits with thrown error.

## Steps to Reproduce

1. Setup the manifest from below.
2. Run manifest with following command:
```
npm run ie -- --manifest /Users/mariamkhalatova/Projects/UK/if/manifests/examples/basic.yml --output
```

## Manifest File That Generated the Error

```yaml
name: basic
description: a minimal manifest executing a single plugin on a single component for a single timestep
tags:
initialize:
  outputs: ['yaml']
  plugins:
    teads-curve: 
      path: '@grnsft/if-unofficial-plugins'
      method: TeadsCurve
      global-config:
        interpolation: spline
tree:
  children:
    child-0:
      defaults:
        cpu/thermal-design-power: 100
      pipeline:
        - teads-curve
      inputs:
        - timestamp: 2023-07-06T00:00
          duration: 1
          cpu/utilization: 20
        - timestamp: 2023-07-06T00:01
          duration: 1
          cpu/utilization: 80
        - timestamp: 2023-07-06T00:02
          duration: 1
          cpu/utilization: 20

```

## Links to Any Additional Code

```
npm run ie -- --manifest /Users/mariamkhalatova/Projects/UK/if/manifests/examples/basic.yml --output
```

## Runtime Info

macOS, Node 18

**Comment by MariamKhalatova** [comment]
#592

**Comment by MariamKhalatova** [comment]
fixed by #691
